api/desecapi/crypto.py

```diff
@@ -0,0 +1,37 @@
+from base64 import urlsafe_b64encode
+from enum import Enum
+
+from cryptography.fernet import Fernet, InvalidToken
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.kbkdf import CounterLocation, KBKDFHMAC, Mode
+from cryptography.hazmat.backends import default_backend
+from django.conf import settings
+from django.utils.encoding import force_bytes
+
+
+def _derive_urlsafe_key(*, label, context):
+    backend = default_backend()
+    kdf = KBKDFHMAC(algorithm=hashes.SHA256(), mode=Mode.CounterMode, length=32, rlen=4, llen=4,
```

Comment:
Hard-coding things like key-length might lead to trouble later on if you want to change it. Why do you need 32 byte? Could this change?

Reply:
Currently, we use the key exclusively with the [Fernet crypto implementation](https://cryptography.io/en/latest/fernet/#cryptography.fernet.Fernet) which requires a URL-safe base64-encoded 32-byte key. If it ever changes, we can certainly add another kwarg here.

Also, things produced with this key are not persisted at the moment -- the use case is verification codes in things like password reset emails. Those links have a validity window of 12 windows. If all else fails, we could stop issuing such links for 12 hours and perform whatever migration we like (maybe use a shorter expiration window temporarily).